docs/developers/adr/002_structure_for_describing_macroscopic_structure.md

```diff
@@ -0,0 +1,26 @@
+# A Structure for Describing Macroscopic Structure
+
+## Context
+
+The main job of spinW is to calculate the energy levels of a magnetic system over a range of q values. This
+task is relatively self-contained and well defined by just the material's cystallographic and magnetic properties.
+
+However, there are a number of things that are more properly understood as a property of a particular experiment. 
+The main features are to include here is whether or not the crystal is twinned or is a powder.
+
+There is a question about whether it should be the location for the option to apply neutron scattering selection rules.
+
+
+## Decision
+
+Create a class representing the macroscopic structure that is responsible for working out the appropriate inputs for the
+main calculation, and constructing the measurable output.
+
+## Status
+
+Proposed
+

```

Comment:
```suggestion
We will create a base `Crystal` class with derived `TwinnedCrystal` and `PolyCrystal` classes which will have methods for preprocessing the input **Q**-vector list and post-processing the output spectra from `spinwave()` for the case of twinned crystals and powder (polycrystalline) samples. The base class will be used by default for untwinned crystals.

## Status

Accepted
```

Reply:
Looks good, I'd be happy with this. I was wondering to what extent `TwinnedCrystal` can be represented by a collection/container of multiple `Crystal` base classes (perhaps it could still inherit from the `Crystal` base class - or maybe this breaks some rule of polymorphism - I don't know!). I think in general the S(Q,w) would need to be evaluated separately for each twin on the transformed Q-points (e.g. magnetic field may be along different inequivalent crystallographic axes in different twins). I guess the downside is that you might be more efficient to parallelise the spinwave loop over all the Q-points for all twins (as is done for powders) then assign then sum over twins after.

Reply:
Essentially I was thinking of something in `spinwave()` along the lines of

```
q_list = crystal.parse_q(q_list)
Sab, omega = crystal.parse_sqw(calculate(q_list))
```

So in the base class `Crystal`, the `parse_q` and `parse_sqw` are no-ops (I assume that `calculate()` returns a tuple of `Sab, omega`).

Reply:
I'm thinking `Sample` is the right name